src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,338 @@
+    }
+
+    private void checkIfExecutionResultIsEmpty(final CommandResult executionResult) {
+        if (executionResult.getExitCode() != 0 || !executionResult.getStdErr().isEmpty()
+                || executionResult.getStdOut().isEmpty()) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "Failed to fetch partition data. Check partition "
+                    + "name. " + executionResult);
+        }
+    }
+
+    private SlurmQueue getPartitionData(final String resultOutputString) {
+        final String[] nodeDataArray = resultOutputString.split(STANDARD_SLURM_DELIMETER);
+        if (nodeDataArray.length != SCONTROL_OUPUT_SIZE) {
+            throw new GridEngineException(HttpStatus.NOT_FOUND, "Node data is inconsistent: waiting for "
+                    + SCONTROL_OUPUT_SIZE + " fields, but " + nodeDataArray.length + " were fetched.");
+        }
+
+        final List<String> nodeList = parseGroupOfNodes(
+                Arrays.stream(nodeDataArray[SCONTROL_OUPUT_NODES_INDEX].split(COMMA))
+                        .filter(StringUtils::hasText)
+                        .collect(Collectors.toList())
+        );
+
+        final int cpusPerNode = Integer.parseInt(nodeDataArray[SCONTROL_OUPUT_CPUS_INDEX]);
+
+        final List<String> userGroups = Arrays.stream(nodeDataArray[SCONTROL_OUPUT_USERGROUPS_INDEX].split(COMMA))
+                .filter(StringUtils::hasText)
+                .collect(Collectors.toList());
+
+        return SlurmQueue.builder()
+                .partition(nodeDataArray[SCONTROL_OUPUT_PARTNAME_INDEX])
+                .nodelist(nodeList)
+                .cpus(cpusPerNode)
+                .groups(userGroups)
+                .build();
+    }
+
+    private List<Queue> fillQueueNameFromOutput(final List<String> resultOutput) {
+        return ListUtils.emptyIfNull(resultOutput).stream()
+                .map(partitionData -> {
+                    final String[] partitionDataArray = partitionData.split(STANDARD_SLURM_DELIMETER);
+                    if (partitionDataArray.length != SCONTROL_OUPUT_SIZE) {
+                        throw new GridEngineException(HttpStatus.NOT_FOUND, "Partition output data is incorrect");
+                    }
+                    return partitionDataArray[SCONTROL_OUPUT_PARTNAME_INDEX].trim();
+                })
+                .distinct()
+                .map(partitionName -> Queue.builder()
+                        .name(partitionName)
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    private List<Queue> fillQueueData(final List<String> resultOutput) {
+        return ListUtils.emptyIfNull(resultOutput).stream()
+                .map(this::getPartitionData)
+                .map(queueMapper::slurmQueueToQueue)
+                .collect(Collectors.toList());
+    }
+
+    private Map<String, Integer> getNodesDescription(final List<String> nodes, final int cpus) {
+        return nodes.stream()
```

Comment:
It looks like this method duplicates code from `SlurmQueueMapper#mapSlurmSlotsToSlots`. May be we can extract method in `SlurmQueueMapper` and reuse it?

Reply:
done